### PR TITLE
[kie-issues-249] Data Index Events changes for allowing historical data

### PR DESCRIPTION
**Pull request by elguardian**
Issue: https://github.com/kiegroup/kie-issues/issues/249

depends on https://github.com/apache/incubator-kie-drools/pull/5529

### CODE REVIEW

**Review by porcelli — CHANGES_REQUESTED** [review]
file headers should be updated to Apache

File: kogito-codegen-modules/kogito-codegen-processes-integration-tests/src/test/java/org/kie/kogito/codegen/tests/PublishEventIT.java [hunk 1]
```diff
@@ -207,11 +225,14 @@ public void testBasicUserTaskProcess() throws Exception {
         uow.end();
         assertThat(processInstance.status()).isEqualTo(ProcessInstance.STATE_COMPLETED);
         events = publisher.extract();
-        assertThat(events).hasSize(2);
-        body = assertProcessInstanceEvent(events.get(0), "UserTasksProcess", "UserTasksProcess", 2);
-        assertThat(body.getNodeInstances()).hasSize(2).extractingResultOf("getNodeType").contains("HumanTaskNode", "EndNode");
-        assertThat(body.getNodeInstances()).extractingResultOf("getTriggerTime").allMatch(v -> v != null);
-        assertThat(body.getNodeInstances()).extractingResultOf("getLeaveTime").allMatch(v -> v != null);
+
+        assertProcessInstanceEvent(events.get(3), "UserTasksProcess", "UserTasksProcess", 2);
```
Comment (pefernan) [inline kogito-codegen-modules/kogito-codegen-processes-integration-tests/src/test/java/org/kie/kogito/codegen/tests/PublishEventIT.java:229]: @elguardian this test is failing here... I can fix it with this:
```suggestion
        assertNodeInstanceEvent(events.get(3), processInstance.id(), "End Event 1", 1);
```

**Review by pefernan — APPROVED** [review]
+1 as soon as we have some greens :D

**Comment by elguardian** [comment]
build now

**Comment by elguardian** [comment]
build now

**Pull request closed without merging**